python/plugins/processing/script/ScriptTemplateWithDecorators.py

```diff
@@ -0,0 +1,127 @@
+    """
+
+    # Retrieve the feature source and sink. The 'dest_id' variable is used
+    # to uniquely identify the feature sink, and must be included in the
+    # dictionary returned by the processAlgorithm function.
+    source = instance.parameterAsSource(
+        parameters,
+        'INPUT',
+        context
+    )
+
+    # If source was not found, throw an exception to indicate that the algorithm
+    # encountered a fatal error. The exception text can be any string, but in this
+    # case we use the pre-built invalidSourceError method to return a standard
+    # helper text for when a source cannot be evaluated
+    if source is None:
+        raise QgsProcessingException(instance.invalidSourceError(parameters,
+                                                                 'INPUT'))
+
+    (sink, dest_id) = instance.parameterAsSink(
+        parameters,
+        'OUTPUT',
+        context,
+        source.fields(),
+        source.wkbType(),
+        source.sourceCrs()
+    )
+
+    # Send some information to the user
+    feedback.pushInfo('CRS is {}'.format(source.sourceCrs().authid()))
+
+    # If sink was not created, throw an exception to indicate that the algorithm
+    # encountered a fatal error. The exception text can be any string, but in this
+    # case we use the pre-built invalidSinkError method to return a standard
+    # helper text for when a sink cannot be evaluated
+    if sink is None:
+        raise QgsProcessingException(instance.invalidSinkError(parameters,
+                                                               'OUTPUT'))
+
+    # Compute the number of steps to display within the progress bar and
+    # get features from source
+    total = 100.0 / source.featureCount() if source.featureCount() else 0
+    features = source.getFeatures()
+
+    for current, feature in enumerate(features):
+        # Stop the algorithm if cancel button has been clicked
+        if feedback.isCanceled():
+            break
+
+        # Add a feature in the sink
+        sink.addFeature(feature, QgsFeatureSink.FastInsert)
+
+        # Update the progress bar
+        feedback.setProgress(int(current * total))
+
+    # To run another Processing algorithm as part of this algorithm, you can use
+    # processing.run(...). Make sure you pass the current context and feedback
+    # to processing.run to ensure that all temporary layer outputs are available
+    # to the executed algorithm, and that the executed algorithm can send feedback
+    # reports to the user (and correctly handle cancellation and progress reports!)
+    if False:
+        buffered_layer = processing.run("native:buffer", {
+            'INPUT': dest_id,
+            'DISTANCE': 1.5,
+            'SEGMENTS': 5,
+            'END_CAP_STYLE': 0,
+            'JOIN_STYLE': 0,
+            'MITER_LIMIT': 2,
+            'DISSOLVE': False,
+            'OUTPUT': 'memory:'
+        }, context=context, feedback=feedback)['OUTPUT']
```

Comment:
Can we remove this one from the example? From discussion in https://github.com/qgis/QGIS/pull/46851 we should separate this part out into its own (more explanatory) example so that we can keep the other templates nice and straightforward.

Reply:
Sure! For me it's also too much for a first algorithm.
I just removed that block in both templates.

Now that you mention it, it would be great to have in the future other templates as well, like "Algorithm with edit features 'in place' support".

Reply:
@gacarrillor 
before we remove it from the original template, would you consider adding a new template with this in place? (And not disabled with a "if False" condition!). It'd be great to have a good, (useful!) example with calling other tools...

Reply:
@nyalldawson, yes, I think it's possible. However, it will take me some extra time because I'm now preparing other PRs on my spare time (I came with some ideas from the hackfest in Firenze).

Do you think we could proceed and merge this PR either as it is or reverting the previous commit (the one removing the 'advanced' block in both templates), or do you think we should really wait and include here also the new script calling other algorithms?